exchange/bitswap/bitswap.go

```diff
@@ -151,7 +151,10 @@ func (bs *bitswap) GetBlock(parent context.Context, k u.Key) (*blocks.Block, err
 	}
 
 	select {
-	case block := <-promise:
+	case block, ok := <-promise:
+		if !ok {
+			return nil, errors.New("promise channel was closed")
+		}
```

Comment:
if this really is a ctx cancellation, then it's probably best to use the ctx error. i suggest something like:

```go
  case block, ok := <-promise:
    if !ok {
      select {
      case <-ctx.Done():
        return nil, ctx.Err()
      default:
        return nil, errors.New("promise channel closed unexpectedly")
      }
    }
```